pkg/kube/client.go

```diff
@@ -506,10 +506,19 @@ func (c *client) RunAndWait(stop <-chan struct{}) {
 func (c *client) GetKubernetesVersion() (*kubeVersion.Info, error) {
 	c.versionOnce.Do(func() {
 		v, err := c.Discovery().ServerVersion()
-		c.version = v
-		c.versionErr = err
+		if err == nil {
+			c.version = v
+		}
 	})
-	return c.version, c.versionErr
+	if c.version != nil {
+		return c.version, nil
+	}
+	// Initial attempt failed, retry on each call to this function
+	v, err := c.Discovery().ServerVersion()
+	if err != nil {
```

Comment:
Is this a bug?  Should be `if err == nil {` ?

Reply:
Good catch https://github.com/istio/istio/pull/41175